### PR TITLE
Add QuadrigaCX exchange and ticker

### PR DESCRIPTION
**Pull request by RafaelTaranto**

### CODE REVIEW

**Comment by RafaelTaranto** [comment]
[QuadrigaCX ticker/trader for BCH, ETH, BTC, LTC](https://trello.com/c/ibJ7jM4Q/325-quadrigacx-ticker-trader-for-bch-eth-btc-ltc)

File: public/elm.js [hunk 1]
```diff
@@ -5759,11 +5759,17 @@ var _elm_lang$core$Platform$Router = {ctor: 'Router'};
 
 var _NoRedInk$elm_decode_pipeline$Json_Decode_Pipeline$decode = _elm_lang$core$Json_Decode$succeed;
 var _NoRedInk$elm_decode_pipeline$Json_Decode_Pipeline$resolve = _elm_lang$core$Json_Decode$andThen(_elm_lang$core$Basics$identity);
-var _NoRedInk$elm_decode_pipeline$Json_Decode_Pipeline$custom = _elm_lang$core$Json_Decode$map2(
-	F2(
-		function (x, y) {
-			return y(x);
-		}));
+var _NoRedInk$elm_decode_pipeline$Json_Decode_Pipeline$custom = F2(
```
Comment (RafaelTaranto) [inline public/elm.js]: Not sure what triggered this change

File: public/elm.js [hunk 2]
```diff
@@ -5795,7 +5801,15 @@ var _NoRedInk$elm_decode_pipeline$Json_Decode_Pipeline$optionalDecoder = F3(
 					return _elm_lang$core$Json_Decode$fail(_p2._0);
 				}
 			} else {
-				return _elm_lang$core$Json_Decode$succeed(fallback);
+				var _p3 = A2(
```
Comment (RafaelTaranto) [inline public/elm.js]: Nor this one

File: lib/plugins/common/quadrigacx.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    .createHmac('sha256', Buffer.from(config.secret))
+    .update(msg)
+    .digest('hex')
+    .toLowerCase()
+
+  const signedData = _.merge(data, {
+    key: config.key,
+    signature: signature,
+    nonce: nonce
+  })
```
Comment (joshmh) [inline lib/plugins/common/quadrigacx.js]: Could be more concise:

```javascript
_.merge(data, {
    key: config.key,
    signature,
    nonce
  })
```

File: lib/plugins/common/quadrigacx.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+  return request(path, 'POST', signedData)
+}
+
+function buildMarket (fiatCode, cryptoCode) {
+  if (!_.includes(cryptoCode, ['BTC', 'ETH', 'LTC', 'BCH'])) {
+    throw new Error('Unsupported crypto: ' + cryptoCode)
```
Comment (joshmh) [inline lib/plugins/common/quadrigacx.js]: You can use backticks here:

```javascript
throw new Error(`Unsupported crypto: ${cryptoCode}`)
```

File: lib/plugins/exchange/quadrigacx/quadrigacx.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+function trade (type, account, cryptoAtoms, fiatCode, cryptoCode) {
+  try {
+    const market = common.buildMarket(fiatCode, cryptoCode)
+    const options = {
+      book: market,
+      amount: coinUtils.toUnit(cryptoAtoms, cryptoCode).toFixed(8)
+    }
+
+    return common.authRequest(account, '/' + type, options)
```
Comment (joshmh) [inline lib/plugins/exchange/quadrigacx/quadrigacx.js]: Will this properly catch promise errors? Probably simpler to make the whole function promise-based:

```javascript
return Promise.resolve()
  .then(() => {
    const market = common.buildMarket(fiatCode, cryptoCode)
    const options = {
      book: market,
      amount: coinUtils.toUnit(cryptoAtoms, cryptoCode).toFixed(8)
    }
    
    return common.authRequest(account, '/' + type, options)
  })
```

File: lib/plugins/common/quadrigacx.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
+'use strict'
```
Comment (joshmh) [inline lib/plugins/common/quadrigacx.js]: Not necessary.

**Review by joshmh — CHANGES_REQUESTED** [review]

**Pull request merged**